test/inductor/test_compiled_optimizers.py

```diff
@@ -379,6 +380,49 @@ def fn():
         self.assertTrue(p_ref() is None)
         gc.enable()
 
+    @requires_cuda
+    def test_disabled_lbfgs(self):
+        torch._dynamo.reset()
+        torch._inductor.metrics.reset()
+        input = torch.ones([10, 10], device="cuda")
+        model = torch.nn.Sequential(
+            *[torch.nn.Linear(10, 10, device="cuda") for _ in range(2)]
+        )
+        model(input).sum().backward()
+        opt = LBFGS(model.parameters(), lr=0.1)
+
+        @torch.compile()
+        def fn():
+            opt.step(closure=lambda: model(input).sum())
```

Comment:
```suggestion
            opt.step(closure=lambda: model(input).sum())
```
The closure in real life also normally computes the gradients + calls the backward()